openedx/features/content_type_gating/tests/test_access.py

```diff
@@ -368,3 +375,93 @@ def test_xblock_handlers(self, xblock_type, xblock_name, user, status_code):
         self.client.login(username=self.users[user].username, password=TEST_PASSWORD)
         response = self.client.post(url)
         self.assertEqual(response.status_code, status_code)
+
+
+@ddt.ddt
+@override_waffle_flag(CONTENT_TYPE_GATING_FLAG, True)
+@override_settings(FIELD_OVERRIDE_PROVIDERS=(
+    'openedx.features.content_type_gating.field_override.ContentTypeGatingFieldOverride',
+))
+class TestConditionalContentAccess(TestConditionalContent):
+    @classmethod
+    def setUpClass(cls):
+        super(TestConditionalContentAccess, cls).setUpClass()
+        cls.factory = RequestFactory()
+    
+    def setUp(self):
+        super(TestConditionalContentAccess, self).setUp()
+
+        CourseModeFactory.create(course_id=self.course.id, mode_slug='audit')
+        CourseModeFactory.create(course_id=self.course.id, mode_slug='verified')
+
+        # Audit users have already been created. Now create verified users, and partition.
+        self.student_verified_a = UserFactory.create(username='student_verified_a', email='student_verified_a@example.com')
+        CourseEnrollmentFactory.create(user=self.student_verified_a, course_id=self.course.id, mode='verified')
+        self.student_verified_b = UserFactory.create(username='student_verified_b', email='student_verified_b@example.com')
+        CourseEnrollmentFactory.create(user=self.student_verified_b, course_id=self.course.id, mode='verified')
+
+        # Use this to put users into content gating groups
+        UserCourseTagFactory(
+            user=self.student_verified_a,
+            course_id=self.course.id,
+            key='xblock.partition_service.partition_{0}'.format(self.partition.id),
+            value=str(self.user_partition_group_a)
+        )
+        UserCourseTagFactory(
+            user=self.student_verified_b,
+            course_id=self.course.id,
+            key='xblock.partition_service.partition_{0}'.format(self.partition.id),
+            value=str(self.user_partition_group_b)
+        )
+
+        self.block_a = ItemFactory.create(
+            parent=self.vertical_a,
+            category='problem',
+            display_name='problem_a',
+            graded=True,
+            weight=1,
```

Comment:
i don't think you need weight on either of these.  
you also may not need display_name depending on if/how you use ddt

Reply:
okay, I can change it. I had a little bit of trouble tracing this function, and figuring out what parameters were necessary.  Do you mind pointing me to this information?

Reply:
for me in this case, and in similar cases, where I am passing parameters to a function that may not be required, I will typically delete them one at a time and rerun the test until I understand which ones are required.  
(or put a debugger and make the function call in the console multiple times to understand what the parameters are doing)